identity/verifier_test.go

```diff
@@ -0,0 +1,42 @@
+package identity
+
+import (
+	"crypto/ecdsa"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestVerifyFunctionReturnsTrueWhenSignatureIsCorrect(t *testing.T) {
+	message := []byte("I am message!")
+
+	privateKey, err := crypto.GenerateKey()
+	assert.NoError(t, err)
+
+	hash := messageHash(message)
+	signature, err := crypto.Sign(hash, privateKey)
+	assert.NoError(t, err)
+
+	identity := toIdentity(privateKey)
+	assert.True(t, NewVerifier(identity).Verify(message, signature))
+}
+
+func TestVerifyFunctionReturnsFalseWhenSignatureIsIncorrect(t *testing.T) {
+	message := []byte("I am message!")
+	privateKey, err := crypto.GenerateKey()
+	assert.NoError(t, err)
+
+	signature, err := crypto.Sign(crypto.Keccak256(message), privateKey)
```

Comment:
If I'm not mistaken, we could re-use `messageHash` here instead of `crypto.Keccak256`, as in `TestVerifyFunctionReturnsTrueWhenSignatureIsCorrect`. That it would be easier to see the difference between these two tests - only different parts will be different.

Reply:
Missed that. Good point.